compiler/infra/BitVector.hpp

```diff
@@ -163,6 +163,28 @@ class TR_BitVector
    TR_BitVector() : _numChunks(0), _chunks(NULL), _firstChunkWithNonZero(0), _lastChunkWithNonZero(-1), _growable(growable), _region(0) { }
    TR_BitVector(TR::Region &region) : _numChunks(0), _chunks(NULL), _firstChunkWithNonZero(0), _lastChunkWithNonZero(-1), _growable(growable), _region(&region) { }
 
+   /**
+    * @brief Constructor to create a new BitVector by reading serialized data from the memory buffer
+    * @param [in] buffer Memory buffer containing serialized BitVector
+    */
+   TR_BitVector(uint8_t * &buffer)
+      {
+      _firstChunkWithNonZero = *(int32_t *)buffer;
+      buffer += sizeof(_firstChunkWithNonZero);
+      _lastChunkWithNonZero = *(int32_t *)buffer;
+      buffer += sizeof(_lastChunkWithNonZero);
+      _numChunks = *(int32_t *)buffer;
+      _chunks = NULL;
+      if (_numChunks > 0)
+         {
+         _chunks = (chunk_t*) TR_Memory::jitPersistentAlloc(_numChunks * sizeof(*_chunks), TR_Memory::BitVector);
+         memcpy(_chunks, buffer, _numChunks * sizeof(_chunks[0]));
+         buffer += (_numChunks * sizeof(_chunks[0]));
+         }
```

Comment:
It would be better to use a struct for these fields, as you don't have to worry about getting the order exactly correct nor having to deal with any subtleties. Something like:

```
struct TR_SerializedBitVector
   {
   int32_t _firstChunkWithNonZero;
   int32_t _lastChunkWithNonZero;
   int32_t _numChunks;
   }
...
   TR_BitVector(uint8_t * &buffer)
      {
      TR_SerializedBitVector *data = (TR_SerializedBitVector *)buffer;
      _firstChunkWithNonZero = data->_firstChunkWithNonZero;
      _lastChunkWithNonZero = data->_lastChunkWithNonZero;
      _numChunks = data->_numChunks;
     ...
     uint8_t *serializedChunks = buffer + sizeof(TR_SerializedBitVector);
     memcpy(_chunks, serializedChunks, _numChunks * sizeof(_chunks[0]));
     buffer = serializedChunks + (_numChunks * sizeof(_chunks[0]));
     ...
```
Also, put `_numChunks * sizeof(*_chunks)` into a variable since it's used three times.